src/map/latent_effect_container.cpp

```diff
@@ -659,15 +659,19 @@ void CLatentEffectContainer::CheckLatentsWeather()
 void CLatentEffectContainer::CheckLatentsWeather(uint16 weather)
 {
     ProcessLatentEffects([this, weather](CLatentEffect& latent) {
-        if (latent.GetConditionsID() == LATENT::WEATHER_ELEMENT)
+        auto owner = dynamic_cast<CBattleEntity*>(m_POwner);
+        if (owner != nullptr)
```

Comment:
if (auto POwner = dynamic_cast<CBattleEntity*>(m_POwner))

Reply:
As a thought process to try and take on board:

If there are two classes like `Entity` and `ContainerOwnedByEntity`, where `ContainerOwnedByEntity` is constructed with a reference or a pointer back to the entity that owns it (like `m_POwner`) - there is no scenario where that container should exist or have its functions called where `m_POwner` isn't valid. We have deep deep lifetime issues if this is the case, and I think it's _extremely unsafe_ to try and sweep it under the rug, even if it's a pain in the ass to try and track down the root cause.